ci/windows_dev_env/windows_deps_headless_installer.py

```diff
@@ -0,0 +1,373 @@
+        copy(tmpdir+"\\cuda\\lib\\x64\\cudnn.lib","C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v9.2\\lib\\x64")
+    logging.info("cuDNN install complete")
+
+
+def install_nvdriver():
+    logging.info("Installing Nvidia Display Drivers...")
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_file = download(DEPS['nvdriver'])
+        with zipfile.ZipFile(local_file, 'r') as zip:
+            zip.extractall(tmpdir)
+        run_command(tmpdir + "\\setup.exe /n /s /noeula /nofinish")
+    logging.info("NVidia install complete")
+
+
+def install_cuda():
+    # CUDA 9.2 and patches
+    logging.info("Installing CUDA 9.2 and Patches...")
+    cuda_9_2_file_path = download('https://developer.nvidia.com/compute/cuda/9.2/Prod2/network_installers2/cuda_9.2.148_win10_network')
+    run_command("PowerShell Rename-Item -Path {} -NewName \"{}.exe\"".format(cuda_9_2_file_path, cuda_9_2_file_path.split('\\')[-1]), shell=True)
+    cuda_9_2_file_path = cuda_9_2_file_path + '.exe'
+    run_command(cuda_9_2_file_path \
+    + ' -s nvcc_9.2' \
+    + ' cuobjdump_9.2' \
+    + ' nvprune_9.2' \
+    + ' cupti_9.2' \
+    + ' gpu_library_advisor_9.2' \
+    + ' memcheck_9.2' \
+    + ' nvdisasm_9.2' \
+    + ' nvprof_9.2' \
+    + ' visual_profiler_9.2' \
+    + ' visual_studio_integration_9.2' \
+    + ' demo_suite_9.2' \
+    + ' documentation_9.2' \
+    + ' cublas_9.2' \
+    + ' cublas_dev_9.2' \
+    + ' cudart_9.2' \
+    + ' cufft_9.2' \
+    + ' cufft_dev_9.2' \
+    + ' curand_9.2' \
+    + ' curand_dev_9.2' \
+    + ' cusolver_9.2' \
+    + ' cusolver_dev_9.2' \
+    + ' cusparse_9.2' \
+    + ' cusparse_dev_9.2' \
+    + ' nvgraph_9.2' \
+    + ' nvgraph_dev_9.2' \
+    + ' npp_9.2' \
+    + ' npp_dev_9.2' \
+    + ' nvrtc_9.2' \
+    + ' nvrtc_dev_9.2' \
+    + ' nvml_dev_9.2' \
+    + ' occupancy_calculator_9.2'
+    )
+
+
+def add_paths():
+    # TODO: Add python paths (python -> C:\\Python37\\python.exe, python2 -> C:\\Python27\\python.exe)
+    logging.info("Adding Windows Kits to PATH...")
+    current_path = run_command("PowerShell (Get-Itemproperty -path 'hklm:\\system\\currentcontrolset\\control\\session manager\\environment' -Name Path).Path")
+    logging.debug("current_path: {}".format(current_path))
+    new_path = current_path + ";C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.16299.0\\x86;C:\\Program Files\\OpenBLAS-windows-v0_2_19\\bin"
+    logging.debug("new_path: {}".format(new_path))
+    run_command("PowerShell Set-ItemProperty -path 'hklm:\\system\\currentcontrolset\\control\\session manager\\environment' -Name Path -Value '" + new_path + "'")
+
+
+def has_gpu():
+    gpu_family = {'p2', 'p3', 'g4dn', 'p3dn', 'g3', 'g2', 'g3s'}
+    def instance_family():
+        return urllib.request.urlopen('http://instance-data/latest/meta-data/instance-type').read().decode().split('.')[0]
+    try:
+        return instance_family() in gpu_family
+    except:
+        return False
```

Comment:
Can we log the exception?

Reply:
why log that if the script is executed outside aws? it's a valid use case.

Reply:
Yeah, I'm thinking of cases like outside AWS